p2p/sql_stmt.go

```
@@ -2,36 +2,57 @@ package p2p
 
 import (
 	"database/sql"
+	"fmt"
 	"math"
+	"math/big"
+	"strings"
+	"time"
 
 	_ "github.com/go-sql-driver/mysql"
 	"github.com/teamnsrg/go-ethereum/log"
 	"github.com/teamnsrg/go-ethereum/p2p/discover"
-	"math/big"
-	"time"
 )
 
 func (srv *Server) initSql() error {
-	if srv.MySQLName != "" {
+	if srv.MySQLName == "" {
+		log.Trace("No sql db connection info provided")
```

It seems like we should fail hard here

---

I left it this way to allow debugging node-finder's behaviors unrelated to sql database.
If the connection info is not provided at run-time, `srv.DB` will remain NULL and node-finder will simply skip any behavior related to sql database.